### PR TITLE
Remove constant optimizer again.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Review by georgwiese — APPROVED** [review]

**Comment by leonardoalt** [comment]
tests are failing

File: pilopt/src/lib.rs [hunk 1]
```diff
@@ -104,10 +56,10 @@ fn remove_constant_fixed_columns<T: FieldElement>(pil_file: &mut Analyzed<T>) {
             name: _,
             index,
             next: _,
-            poly_id,
+            poly_id: Some(poly_id),
```
Comment (chriseth) [inline pilopt/src/lib.rs:59]: This is needed because we can still have constant references inside the fixed column code.

**Review by leonardoalt — COMMENTED** [review]
why?

**Review by georgwiese — APPROVED** [review]
LGTM!

**Pull request merged**